frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -295,7 +295,11 @@ function ReaderFooter:resetLayout()
 end
 
 function ReaderFooter:getHeight()
-    return self.footer_content:getSize().h
+    if self.footer_content then
+        return self.footer_content:getSize().h
+    else
+        return 0
+    end
```

Comment:
Why was that needed for? (And why now and never before?)

I'm just worried that if it's an ordering issue (some other module calling this before ReaderFooter is set-up), we may end up with a different bottom margin, which may trigger after first load/rendering, on the next painting, a second rendering, or something - or when loading from cache if we start with this being 0 and a smaller bottom margin, and end up with this being correctly non-zero and then a larger bottom margin.)

Reply:
I need to use it because unit test fails:
https://circleci.com/gh/koreader/koreader/3113#tests/containers/1

```
frontend/apps/reader/modules/readerfooter.lua:298: attempt to index field 'footer_content' (a nil value)

stack traceback:
	frontend/apps/reader/modules/readerfooter.lua:298: in function 'getHeight'
	frontend/apps/reader/modules/readertypeset.lua:509: in function 'onSetPageMargins'
	frontend/apps/reader/modules/readertypeset.lua:92: in function 'handleEvent'
	frontend/ui/widget/container/widgetcontainer.lua:88: in function 'propagateEvent'
	frontend/ui/widget/container/widgetcontainer.lua:106: in function 'handleEvent'
	frontend/apps/reader/readerui.lua:385: in function 'init'
	frontend/ui/widget/widget.lua:48: in function 'new'
	spec/front/unit/readerfooter_spec.lua:372: in function <spec/front/unit/readerfooter_spec.lua:360>

[  ERROR   ] spec/front/unit/readerfooter_spec.lua @ 360: Readerfooter module should not schedule auto refresh time task if footer is disabled (149.12 ms)
```

Reply:
I can leave function `ReaderFooter:getHeight()` as was before and change `function ReaderTypeset:onSetPageMargins()` (adding `self.view.footer.settings.disabled` in if)



```
function ReaderTypeset:onSetPageMargins(margins, refresh_callback)
    local left = Screen:scaleBySize(margins[1])
    local top = Screen:scaleBySize(margins[2])
    local right = Screen:scaleBySize(margins[3])
    local bottom
    if self.view.footer.has_no_mode or self.view.footer.reclaim_height or self.view.footer.settings.disabled then
        bottom = Screen:scaleBySize(margins[4])
    else
        bottom = Screen:scaleBySize(margins[4]) + self.view.footer:getHeight()
    end
    self.ui.document:setPageMargins(left, top, right, bottom)
    self.ui:handleEvent(Event:new("UpdatePos"))
    if refresh_callback then
        -- Show a toast on set, with the unscaled & scaled values
        UIManager:show(InfoMessage:new{
            text = T(_([[
    Margins set to:

    horizontal: %1 (%2px)
    top: %3 (%4px)
    bottom: %5 (%6px)

    Tap to dismiss.]]),
            margins[1], left, margins[2], top, margins[4], bottom),
            dismiss_callback = refresh_callback,
        })
    end
    return true
end

```

Reply:
> I need to use it because unit test fails

OK, so it's just because that function was now called in a rarely tested case and it failed.
So, I guess your change is fine.

> I can leave function ReaderFooter:getHeight() as was before and change function ReaderTypeset:onSetPageMargins() (adding self.view.footer.settings.disabled in if)

Well, no, it's fine where you put it. Because I'm sure that the other places where I use `footer:getHeight()` would then fail too if I tested them with footer disabled :) (Even if it looks to me there's no way to set disable=false via the UI, so that unit test is testing something that's never happening/used ?)